packages/block-editor/src/components/block-list/block-list-compact.native.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import BlockListBlock from './block';
+
+function BlockListCompact( props ) {
+	const {
+		marginHorizontal = styles.defaultBlock.marginLeft,
+		marginVertical = styles.defaultBlock.marginTop,
+		rootClientId,
+	} = props;
+	const { blockClientIds } = useSelect(
+		( select ) => {
+			const { getBlockOrder } = select( blockEditorStore );
+			const blockOrder = getBlockOrder( rootClientId );
+
+			return {
+				blockClientIds: blockOrder,
+			};
```

Comment:
The `InnerBlock` component passes the prop `filterInnerBlocks` that is used to filter out inner blocks.
https://github.com/WordPress/gutenberg/blob/1d9c05aa655d6039aca26980c89b1e23b6935598/packages/block-editor/src/components/inner-blocks/index.native.js#L135

Here you can see how it's done in `BlockList` component:
https://github.com/WordPress/gutenberg/blob/1d9c05aa655d6039aca26980c89b1e23b6935598/packages/block-editor/src/components/block-list/index.native.js#L421-L424

I'm wondering if we should also apply the filtering in the compact version, wdyt?

Reply:
> Maybe we could wait and see if we add this simpler block list to other blocks that use inner blocks, what do you think?

Sounds good, especially in the spirit of addressing the issues in the `List` block V2, we could stick to this approach. My main concern about not including this, similar as [my previous comment](https://github.com/WordPress/gutenberg/pull/43431#discussion_r952501419), is that if in the future we use this component in more places, we might forget about the missing functionalities compared to the regular `BlockList`. Maybe we could add a comment at the beginning of the component stating this, wdyt?

Reply:
> Maybe we could add a comment at the beginning of the component stating this, wdyt?

Good idea! I'll add a note 👍